core/src/main/java/com/linecorp/armeria/common/stream/DeferredStreamMessage.java

```diff
@@ -81,10 +93,19 @@ protected void delegate(StreamMessage<T> delegate) {
             }).exceptionally(CompletionActions::log);
         }
 
-        @SuppressWarnings("unchecked")
-        final PendingSubscription<? super T> ps = pendingSubscription;
-        if (ps != null) {
-            subscribeToDelegate(delegate, ps);
+        // There are 4 state in class.
+        // If delegate is called first, then it will match INIT -> DELEGATED.
+        // If subscribe is called first, than it will match SUBSCRIBED -> SUBSCRIBED_AND_DELEGATED.
```

Comment:
- subscribe -> subscribe()
- delegate -> delegate()
- then it will match -> the state will transit from ... to ...